File: 929 - Unique Email Addressed.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int numUniqueEmails(String[] emails) {
        Set<String> s = new HashSet<>();
        for (String email : emails) {
            int at = email.indexOf("@");
            String local = email.substring(0, at), domain = email.substring(at);
            if (local.contains("+")) {
                local = local.substring(0, local.indexOf('+'));
            }
            local = local.replaceAll("\\.", "");
            s.add(local + domain);
        }
        return s.size();
    }
}

File: 929 - Unique Email Addressed.cpp
#include <unordered_set>
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    int numUniqueEmails(vector<string>& emails) {
        unordered_set<string> s;
        for (string email : emails) {
            int at = email.find('@');
            string local = email.substr(0, at), domain = email.substr(at);
            string newlocal = "";
            for (char c : local) {
                if (c == '+') {
                    break;
                } else if (c == '.') {
                    continue;
                } else {
                    newlocal += c;
                }
            }
            string res = newlocal + domain;
            s.insert(res);
        }
        return s.size();
    }
};